test/download_helper.js

```diff
@@ -0,0 +1,257 @@
+      const curlCommand = "curl -OJ url1 \\ &\ncurl -OJ url2"
+      const downloadList = DownloadHelper.generateDownloadList();
+      expect(downloadList == curlCommand).to.be.true;
+    });
+  });
+
+  describe("createButton()", () => {
+    it('should add a button to the end of div.download-titles', async () => {
+      createDomNodes(`
+        <div id="testId" class="download-titles"></div>
+      `);
+
+      dh.createButton();
+      const button = document.querySelector(".downloadall")
+
+      expect(button != null).to.be.true;
+    });
+
+    it('should not replace existing button at the end of div.download-titles', async () => {
+      createDomNodes(`
+        <div id="testId" class="download-titles">
+          <button class="downloadall fake" id="someId"></button>
+        </div>
+      `)
+
+      {
+        const button = document.querySelector(".downloadall")
+        const idAttribute = button.getAttribute("id");
+        expect(idAttribute == "someId").to.be.true;
+      }
+
+      dh.createButton();
+      
+      {
+        const button = document.querySelectorAll(".downloadall")
+        assert(button.length == 1);
+        const idAttribute = button[0].getAttribute("id");
+        expect(idAttribute == "someId").to.be.true;
+      }
+    });
+
+    it('should create specific properties for the button', async () => {
+      createDomNodes(`
+        <div id="testId" class="download-titles"></div>
+      `)
+      
+      dh.createButton();
+
+      const buttonClassName = "downloadall"
+      expect(dh.button.className == buttonClassName).to.be.true;
+
+      const buttonTitleExpected = "Generates a file for automating downloads using 'curl'"
+      expect(dh.button.title == buttonTitleExpected).to.be.true;
+
+      expect(dh.button.disabled).to.be.true;
+
+      expect(dh.button.textContent == "preparing download").to.be.true;
+    });
+  });
+
+  describe("enableButton()", () => {
+    it('button should call functions on click', async () => {
+      DownloadHelper.dateString = sinon.fake.returns("dateString");
+      DownloadHelper.generateDownloadList = sinon.fake.returns("downloadList");
+      DownloadHelper.download = sinon.spy();
+
+      dh.button = document.createElement("button");
+
+      dh.enableButton();
+
+      dh.button.click();
+
+      expect(DownloadHelper.dateString.calledOnce).to.be.true;
+      expect(DownloadHelper.generateDownloadList.calledOnce).to.be.true;
+      
+      expect(DownloadHelper.download.calledOnce).to.be.true;
+      expect(DownloadHelper.download.getCall(0).args[0] == "bandcamp_dateString.txt").to.be.true;
+      expect(DownloadHelper.download.getCall(0).args[1] == preamble + "downloadList").to.be.true;
+    });
+
+    it('button should have specific properties', async () => {
+      dh.button = document.createElement("button");
+
+      dh.enableButton();
+
+      expect(dh.button.disabled).to.be.false;
+      expect(dh.button.textContent == "Download curl File").to.be.true;
```

Comment:
Alternative: `expect(dh.button.textContent).to.equal("Download curl File");`

Same result, but Chai prints out more descriptive messages for different string/bool/object comparisons. Give it a try, see what you like better!